docs/source/building/recipe.rst

```diff
@@ -0,0 +1,673 @@
+.. _build:
+
+conda build Recipe Reference
+============================
+
+Building a package requires a recipe.  A recipe is flat directory which
+contains the following files:
+
+  * ``meta.yaml`` (metadata file)
+  * ``build.sh`` (Unix build script which is executed using bash)
+  * ``bld.bat``  (Windows build script which is executed using cmd)
+  * ``run_test.sh``/``.bat``/``.py``/``.pl`` (optional test file)
+  * patches to the source (optional, see below)
+  * other resources, which are not included in the source and cannot be
+    generated by the build scripts.
+
+The same recipe should be used to build a package on all platforms.
+
+When building a package, the following steps are invoked:
+
+  1. read the metadata
+  2. download the source (into a cache)
+  3. extract the source in a *source directory*
+  4. apply the patches
+  5. create a *build environment* (build dependencies are installed here)
+  6. run the actual build script. The current working directory is the source
+     directory with environment variables set. The build script installs into
+     the build environment
+  7. do some necessary post processing steps: shebang, rpath, etc.
+  8. add conda metadata to the build environment
+  9. package up the new files in the build environment into a conda package
+  10. test the new conda package:
+      - create a *test environment* with the package (and its dependencies)
+      - run the test scripts
+
+There are example recipes for many conda packages in the `conda-recipes
+<https://github.com/continuumio/conda-recipes>`_ repo.
+
+The :ref:`conda skeleton <skeleton_ref>` command can help to make skeleton
+recipes for common repositories, such as
+`PyPI <https://pypi.python.org/pypi>`_.
+
+
+The meta.yaml file
+------------------
+
+
+All the metadata in the recipe is specified in the ``meta.yaml`` file. All
+sections are optional except for package/name and package/version.
+
+.. code-block:: yaml
+
+
+    package:
```

Comment:
We should no longer have this giant pseudo-meta.yaml. If you want, put a real meta.yaml here (like one of the ones from the build tutorials).